procrastinate/aiopg_connector.py

```diff
@@ -39,6 +40,45 @@ async def wrapped(*args, **kwargs):
     return wrapped
 
 
+def wrap_query_exceptions(coro: CoroutineFunction) -> CoroutineFunction:
+    """
+    Detect aiopg OperationalError's with a "server closed the connection unexpectedly"
+    message and retry a number of times.
+
+    This is to handle the case where the database connection (obtained from the pool)
+    was actually closed by the server. In this case, aiopg raises an OperationalError
+    with a "server closed the connection unexpectedly" message (and no pgcode) when the
+    connection is used for issuing a query. What we do is retry when an OperationalError
+    is raised, and until max_tries is reached.
+
+    max_tries is set to the max pool size + 1 (maxsize + 1) to handle the case where all
+    the connections we have in the pool were closed by the server.
+    """
+
+    @functools.wraps(coro)
+    async def wrapped(*args, **kwargs):
+        final_exc = None
+        max_tries = (
+            args[0]._pool.maxsize + 1 if args and getattr(args[0], "_pool", None) else 1
```

Comment:
I wonder if this would be easier to understand as:
```
try:
	max_tries = args[0]._pool.maxsize + 1
except Exception:
	max_tries = 1
```

Reply:
I have no preference.

Reply:
then I vote for the change :)